### PR TITLE
Feature enum sources

### PR DESCRIPTION
**Pull request by grburgess**
Switches from using strings as source constants to an enum struct. Better safety.

### CODE REVIEW

File: astromodels/core/model_parser.py [hunk 1]
```diff
@@ -346,15 +347,15 @@ def __init__(self, source_name, source_definition):
...
 
             self._parsed_source = self._parse_point_source(source_definition)
 
-        elif source_type == EXTENDED_SOURCE:
+        elif source_type == SourceType.EXTENDED_SOURCE.value:
 
             self._parsed_source = self._parse_extended_source(source_definition)
 
-        elif source_type == PARTICLE_SOURCE:
+        elif source_type == SourceType.PARTICLE_SOURCE.value:
```
Comment (BjoernBiltzinger) [inline astromodels/core/model_parser.py:359]: I think this .value is not needed in lines 350, 354 and 358

**Review by BjoernBiltzinger — COMMENTED** [review]
Very nice.

File: astromodels/sources/point_source.py [hunk 2]
```diff
@@ -29,6 +29,8 @@
 log = setup_logger(__name__)
 
 
+log = setup_logger(__name__)
```
[the remark this answers is not in the document]
Comment (grburgess) [reply astromodels/sources/point_source.py:32]: thanks!

File: astromodels/sources/point_source.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 from astromodels.core.spectral_component import SpectralComponent
 from astromodels.core.tree import Node
 from astromodels.core.units import get_units
-from astromodels.sources.source import Source, POINT_SOURCE
+from astromodels.sources.source import Source, SourceType
```
[the remark this answers is not in the document]
Comment (grburgess) [reply astromodels/sources/point_source.py:18]: I think it does not matter as it is just a class

File: astromodels/core/model_parser.py [hunk 4]
```diff
@@ -319,16 +318,18 @@ def __init__(self, source_name, source_definition):
...
+            
+            
         except IndexError:  # pragma: no cover
 
             raise ModelSyntaxError(
                 "Don't recognize type for source '%s'. "
                 "Valid types are '%s', '%s' or '%s'."
-                % (source_name, POINT_SOURCE, EXTENDED_SOURCE, PARTICLE_SOURCE)
+                % (source_name, SourceType.POINT_SOURCE, SourceType.EXTENDED_SOURCE, SourceType.PARTICLE_SOURCE)
             )
```
[the remark this answers is not in the document]
Comment (grburgess) [reply astromodels/core/model_parser.py:334]: done

File: astromodels/core/model_parser.py (hunk 1 above)
Comment (grburgess) [reply astromodels/core/model_parser.py:359]: its need for string checking unfortunately

File: astromodels/sources/point_source.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (BjoernBiltzinger) [reply astromodels/sources/point_source.py:19]: Yes I just thought the other way is a bit cleaner. But yes, I think both will work :)

File: astromodels/core/model_parser.py (hunk 1 above)
Comment (BjoernBiltzinger) [reply astromodels/core/model_parser.py:359]: Ah okay

**Review by BjoernBiltzinger — APPROVED** [review]
I think we can merge this now

**Pull request merged**